src/libs/antares/inifile/inifile.cpp

```diff
@@ -231,17 +234,22 @@ bool IniFile::readStream(std::istream& in_stream)
     return true;
 }
 
-bool IniFile::open(const AnyString& filename, bool warnings)
+bool IniFile::open(const std::string& filename, bool warnings)
+{
+    fs::path filepath = filename;
+    return open(filepath, warnings);
+}
+
+bool IniFile::open(const fs::path& filename, bool warnings)
 {
     clear();
-    filename_ = filename; // storing filename for further use
-    std::string filePath = filename.to<std::string>();
+    filename_ = filename.string(); // storing filename for further use
```

Comment:
This conversion is not necessary for `std::ifstream::ifstream` since it accepts `std::filesystem::path`, see https://en.cppreference.com/w/cpp/io/basic_ifstream/basic_ifstream

Reply:
filename_ is a std::string from the Inifile class
it's used in inifile.hxx

explicit conversion is needed for windows compiler 
https://github.com/AntaresSimulatorTeam/Antares_Simulator/actions/runs/9266746246/job/25491625889